pyflagser/flagser.py

```diff
@@ -1,40 +1,153 @@
+          Euler characteristic.
+
+    Notes
+    -----
+    The input graphs cannot contain self-loops, i.e. edges that start and end
+    in the same vertex, therefore diagonal elements of the input adjacency
+    matrix will be ignored.
+
+    References
+    ----------
+    .. [1] D. Luetgehetmann, "Documentation of the C++ flagser library";
+           `GitHub: <https://github.com/luetge/flagser/blob/master/docs/\
+           documentation_flagser.pdf>`_.
+
+    """
+    # Handle default parameters
+    if max_dimension == np.inf:
+        _max_dimension = -1
+    else:
+        _max_dimension = max_dimension
+
+    if approximation is None:
+        _approximation = -1
+    else:
+        _approximation = approximation
+
+    # All edge filtrations are equivalent in the static case
+    _filtration = 'max'
+
+    # Extract vertices and edges
+    vertices, edges = _extract_unweighted_graph(adjacency_matrix)
+
+    # Call flagser binding
+    homology = compute_homology(vertices, edges, min_dimension, _max_dimension,
+                                directed, coeff, _approximation, _filtration)
+
+    # Creating dictionary of return values
+    out = dict()
+    out['betti'] = homology[0].get_betti_numbers()
+    out['cell_count'] = homology[0].get_cell_count()
+    out['euler'] = homology[0].get_euler_characteristic()
+    return out
+
+
+def flagser_weighted(adjacency_matrix, max_edge_weight=None, min_dimension=0,
+                     max_dimension=np.inf, directed=True, filtration="max",
+                     coeff=2, approximation=None):
+    """Compute persistent homology of a directed/undirected filtered flag
+    complex.
+
+    From an adjacency_matrix and a filtration algorithm construct a filtered
+    flag complex as a sequence of its cells associated to their filtration
+    values and compute its persistent homology.
+
+    Parameters
+    ----------
+    adjacency_matrix : 2d ndarray or scipy.sparse matrix of shape
+        (n_vertices, n_vertices), required
+        Matrix representation of a directed/undirected weighted graph. Diagonal
+        elements are vertex weights. The way zero values are handled depends on
+        the format of the matrix. If the matrix is a dense ``numpy.ndarray``,
+        zero values denote zero-weighted edges. If the matrix is a sparse
+        ``scipy.sparse`` matrix, explicitly stored off-diagonal zeros and all
+        diagonal zeros denote zero-weighted edges. Off-diagonal values that
+        have not been explicitely stored are treated by ``scipy.sparse`` as
+        zeros but will be understood as infinitely-valued edges, i.e., edges
+        absent from the filtration.
+
+    max_edge_weight : int or float or ``None``, optional, default: ``None``
+        Maximum edge weight to be considered in the filtration. All edge
+        weights greater than that value will be considered as
+        infinitely-valued, i.e., absent from the filtration. If ``None``, it is
```

Comment:
Depending on what we decide about below, we might have to change this line too.

Reply:
The sentence "If ``None``, [...]" no longer accurately describes what happens. Wouldn't it be enough to add "If not ``None``" to either the first or second sentence? Then the reader would assume ``None`` means there is no maximum weight.